test/tests/outputs/iterative/tests

```diff
@@ -85,4 +85,15 @@
     max_parallel = 1
     max_threads = 1
   [../]
+  [./csv_check_steps]
+    # Test that outputs are done during the specified interval
+    type = 'CSVDiff'
+    input = 'iterative_csv_check_step.i'
+    csvdiff = 'iterative_csv_check_step_out.csv'
+
+    # This test relies on number of residuals being the same, so run it serially
```

Comment:
I really don't like this regression test. Number of residuals is one of the worst quantities to test for because it can change so easily (i.e. parallel, threaded, even PETSc updates). I'd suggest an even more simple test, like "simple_transient_diffusion" where we won't have to carry these restrictions.

Reply:
I too thought that was strange, but I was just copying what had already been done. I would suggest changing all the tests here that use residual counts, but that might be better for a different PR.